runtime/lua/vim/lsp/completion.lua

```diff
@@ -191,7 +191,12 @@ local function get_completion_word(item, prefix, match)
     end
   elseif item.textEdit then
     local word = item.textEdit.newText
-    return word:match('^(%S*)') or word
+    local whitespaceTrimmedWord = word:match('^(%S*)') or word
+    if whitespaceTrimmedWord ~= '' then
+      return whitespaceTrimmedWord
+    else
+      return item.label
```

Comment:
I don't think that falling back to the label here is correct. The docs of [`CompletionItem`](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#completionItem) don't mention falling back to the label if the text edit text is empty.

Note how we fallback to the label when `insertText` is empty, as that is indeed documented on the protocol. However that doesn't seem to be the case here.